src/Core/Module/ModuleManager.php

```diff
@@ -182,6 +182,8 @@ public function upgrade(string $name, $source = null): bool
             $handler->handle($source);
         }
 
+        $this->hookManager->disableHooksForModule($this->moduleDataProvider->getModuleIdByName($name));
```

Comment:
Any change this will cause issues if the module being updated regitered `actionBeforeUpgradeModule` ?

Reply:
On the contrary, I believe this PR is likely to _avoid_ problems due to `actionBeforeUpgradeModule`.

Today, if a module ABC is plugged on `actionBeforeUpgradeModule` then we enter the crazy situation where
- the ABC module is being updated -> this means change of PHP files, cache cleared, new config loaded (like Symfony services definitions), SQL scripts run...
- the ABC module is also called by hook `actionBeforeUpgradeModule`

It's the serpent eating its own tail 😅 and is causing crazy behaviors.

This PR un-plugs the hooks of module ABC before launching the hook. Then the serpent cannot eat its own tail.

From my point of view, we should never have this situation:
> - the ABC module is being updated -> this means change of PHP files, cache cleared, new config loaded (like Symfony services definitions), SQL scripts run...
- the ABC module is also called by hook `actionBeforeUpgradeModule`

This is not a reliable situation.

However what makes sense is
- module ABC is plugged on `actionBeforeUpgradeModule` hook
- module XYZ is being updated -> hook is triggered and module ABC catches the hook and reacts